runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaRunner.java

```diff
@@ -107,7 +107,7 @@ public PortablePipelineResult runPortablePipeline(RunnerApi.Pipeline pipeline) {
 
   @Override
   public SamzaPipelineResult run(Pipeline pipeline) {
-    SplittableParDo.validateNoPrimitiveReads(pipeline);
+    SplittableParDo.convertReadBasedSplittableDoFnsToPrimitiveReadsIfNecessary(pipeline);
```

Comment:
To clarify: this change makes it so instead of rejecting primitive reads the function is a noop?

Reply:
Its a noop if the experiments aren't set. If the experiments are set then `Read.Bounded` and `Read.Unbounded` get converted into `PrimitiveBoundedRead` and `PrimitiveUnboundedRead` respectively.

Reply:
For a runner that previously rejected all primitive reads, isn't it better to leave that runner alone and still reject all primitive reads?

(here and elsewhere)